src/lib/extra-firmware.ts

```diff
@@ -0,0 +1,77 @@
+import { TypedError } from 'typed-error';
+
+import type ConfigJsonConfigBackend from '../config/configJson';
+import { InternalInconsistencyError, isNotFoundError } from './errors';
+import log from './supervisor-console';
+import { docker } from './docker-utils';
+import constants from './constants';
+
+export const EXTRA_FIRMWARE_VOLUME_NAME = 'extra-firmware';
+
+export class ExtraFirmwareVolumeError extends TypedError {
+	constructor() {
+		super(`Found extra firmware volume: ${EXTRA_FIRMWARE_VOLUME_NAME}`);
+	}
+}
+
+export const initialize = async (configJson: ConfigJsonConfigBackend) => {
+	log.info('Configuring extra firmware volume');
+	// Create volume - volume will be empty unless user includes
+	// appropriate linux-firmware block in their docker-compose.
+	// The OS is configured to use the volume path on host for firmware,
+	// but it's on the user to include the firmware block as well as reboot.
+	await create();
```

Comment:
It makes me uneasy that we create the volume here. I think if this crashes it will bring the supervisor down right?? How about doing it via a step like `ensureSupervisorNetwork` does? 

https://github.com/balena-os/balena-supervisor/blob/master/src/compose/application-manager.ts#L214

Reply:
Done! d864dc18a68944bca18a91878cc9e0e548d707b7